### PR TITLE
fix: Show placeholder image when auction results image fails to load

### PR DESCRIPTION
**Pull request by MounirDhahri**
This PR fixes https://artsyproduct.atlassian.net/browse/CX-3594

### Description

https://user-images.githubusercontent.com/11945712/230026087-1d80f84e-2552-4bd0-85ec-283af6281ac1.mov

### PR Checklist

- [x] I have tested my changes on **iOS** and **Android**.
- [x] I hid my changes behind a **[feature flag]**, or they don't need one.
- [x] I have included **screenshots** or **videos**, or I have not changed the UI.
- [x] I have added **tests**, or my changes don't require any.
- [x] I added an **[app state migration]**, or my changes do not require one.
- [x] I have documented any **follow-up work** that this PR will require, or it does not require any.
- [x] I have added a **changelog entry** below, or my changes do not require one.

### To the reviewers 👀

- [ ] I would like **at least one** of the reviewers to **run** this PR on the simulator or device.

<details><summary>Changelog updates</summary>

#### Cross-platform user-facing changes

- Show placeholder image when auction results image fails to load - mounir

#### iOS user-facing changes

-

#### Android user-facing changes

-

#### Dev changes

-

</details>

Need help with something? Have a look at our [docs], or get in touch with us.

[app state migration]: ../blob/main/docs/adding_state_migrations.md
[feature flag]: ../blob/main/docs/developing_a_feature.md
[docs]: ../blob/main/docs/README.md

### CODE REVIEW

File: src/app/Scenes/AuctionResult/AuctionResult.tsx [hunk 1]
```diff
@@ -283,35 +286,49 @@ const AuctionResultImage = ({
...
+  if (!!image.url && !!imageHeight && !!imageWidth && !couldNotLoadImage) {
     return (
       <Flex width="100%" height={CONTAINER_HEIGHT} justifyContent="center" alignItems="center">
-        <Image
+        <FastImage
           style={{ height: imageHeight, width: imageWidth, overflow: "hidden" }}
           source={{ uri: image.url }}
+          onError={() => {
+            addBreadcrumb({
+              message: `Failed to load auction result image`,
```
Comment (MounirDhahri) [inline src/app/Scenes/AuctionResult/AuctionResult.tsx:314]: no need to log the auction result id since it's already available as a navigation breadcrum

**Review by lordkiz — APPROVED** [review]
Niceeee. Looks good. Makes a lot of sense handling these cases.

File: src/app/Components/Lists/AuctionResultListItem.tsx [hunk 2]
```diff
@@ -91,6 +95,13 @@ const AuctionResultListItem: React.FC<Props> = ({
               source={{
                 uri: auctionResult.images.thumbnail.url,
               }}
+              onError={() => {
```
Comment (egdbear) [inline src/app/Components/Lists/AuctionResultListItem.tsx:98]: Not sure how this Component works, but I assume this component tries to load the image and resolves the issue through this prop?

Comment (MounirDhahri) [reply src/app/Components/Lists/AuctionResultListItem.tsx:98]: that's exactly how it works 🙏

Comment (lordkiz) [reply src/app/Components/Lists/AuctionResultListItem.tsx:98]: That is a callback that is called when loading the image errors

**Review by egdbear — APPROVED** [review]
🚀 🚀 🚀

**Pull request merged**